src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -1150,7 +1150,8 @@ def list_container_services(client, resource_group_name=None):
 def create_vm(vm_name, resource_group_name, image,
               size='Standard_DS1', location=None, tags=None, no_wait=False,
               authentication_type=None, admin_password=None, admin_username=getpass.getuser(),
-              ssh_dest_key_path=None, ssh_key_value=None, availability_set=None,
+              ssh_dest_key_path=None, ssh_key_value=None, generate_ssh_keys=False,
```

Comment:
Since you have the flag here, you don't need the "register_extra_cli_argument" entry in params. A regular "register_cli_argument" is sufficient.